### PR TITLE
Preparatory work for management by enarx-keepmgr

### PR DESCRIPTION
**Pull request by MikeCamel**
Currently doesn't actually spawn a keep.
Needs to create Unix socket and communicate with enarx-keepmgr.

Signed-off-by: MikeCamel <mike@p2ptrust.org>

### CODE REVIEW

File: external/enarx-keep-kvm@.service [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Enarx Keep-Loader (KVM)
+Documentation=https://github.com/enarx/
+
+[Service]
+Type=simple
+Environment="ENARX_BACKEND=kvm"
+ExecStart=/home/mike/programming/enarx-keepldr/target/debug/enarx-keepldr kuuid I%
```
Comment (haraldh) [inline external/enarx-keep-kvm@.service:8]: Your personal paths don't belong in the repo :-1:

File: src/main.rs [hunk 2]
```diff
@@ -45,6 +45,10 @@
 //!     $ musl-gcc -static-pie -fPIC -o test test.c
 //!     $ target/debug/enarx-keepldr exec ./test
 //!     Hello World!
+//!
+//!     Or copy (or link) the files in external/*.service to ~/.config/systemd/user/
+//!     $ systemctl --user daemon-reload
+//!     $ systemctl --user start enarx-keep-nil@[Uuid].service
```
Comment (haraldh) [inline src/main.rs:51]: I think this should not run in systemd `--user` mode, but system wide.

**Review by haraldh — CHANGES_REQUESTED** [review]

File: external/enarx-keep-kvm@.service (hunk 1 above)
Comment (MikeCamel) [reply external/enarx-keep-kvm@.service:8]: Fair - what's the canonical place for this sort of thing?

File: src/main.rs (hunk 2 above)
Comment (MikeCamel) [reply src/main.rs:51]: I agree, but there's a balance between allowing multiple tests on a shared system and running as a particular user. I'm not sure what the correct approach is, tbh.

File: external/enarx-keep-kvm@.service (hunk 1 above)
Comment (haraldh) [reply external/enarx-keep-kvm@.service:8]: Well, define a placeholder and document, that the path has to be fixed

File: src/main.rs (hunk 2 above)
Comment (haraldh) [reply src/main.rs:51]: Then, document: "for testing purposes […]"

File: external/enarx-keep-kvm@.service (hunk 1 above)
Comment (haraldh) [reply external/enarx-keep-kvm@.service:8]: Could be  `@BIN@` `@LOG@` ...

**Review by haraldh — CHANGES_REQUESTED** [review]
still failing

**Comment by MikeCamel** [comment]
> still failing

I've fixed the README.md test - now failing on a sallyport issue.   Not me...

**Comment by haraldh** [comment]
> > still failing
> 
> I've fixed the README.md test - now failing on a sallyport issue. Not me...

yay... the benefits of cargo deny all and using cargo nightly... new spurious CI failures  /s

**Comment by npmccallum** [comment]
After a conversation, the `kuuid` command is unnecessary. Closing.

**Pull request closed without merging**